### PR TITLE
Exit and show error message when prelinking

### PR DESCRIPTION
**Pull request by soapy1**
ref: https://github.com/conda/conda/issues/3692

### CODE REVIEW

File: conda/install.py [hunk 1]
```diff
@@ -372,6 +373,10 @@ def run_script(prefix, dist, action='post-link', env_prefix=None):
     env['PKG_NAME'], env['PKG_VERSION'], env['PKG_BUILDNUM'] = \
                 str(dist).rsplit('-', 2)
     if action == 'pre-link':
+        error_and_exit("""
+Package uses a pre-link script. Pre-link scripts are potentially dangerous and highly discouraged.
```
Comment (kalefranz) [inline conda/install.py]: "Package %s uses a..." % dist

File: conda/install.py [hunk 2]
```diff
@@ -372,6 +373,10 @@ def run_script(prefix, dist, action='post-link', env_prefix=None):
     env['PKG_NAME'], env['PKG_VERSION'], env['PKG_BUILDNUM'] = \
                 str(dist).rsplit('-', 2)
     if action == 'pre-link':
+        error_and_exit("""
```
Comment (kalefranz) [inline conda/install.py]: wrap this in `sys.stderr.write()` with a `\n` at the end instead of `error_and_exit()`

**Comment by kalefranz** [comment]
@msarahan @mingwandroid @mcg1969 Are you ok with this?

**Comment by msarahan** [comment]
I am OK with this in concept, but I want to make sure that we understand which of our packages currently use this, so that we can be sure that alternatives actually exist and we aren't left with customers saying "where the hell is a "good" package here?"

**Comment by kalefranz** [comment]
I don't think anything in repo.continuum.io uses this.  Maybe you could ask at the next conda-forge meeting if they know of anything in their repos that does.  Perhaps we could notify the bioconda folks too.  Besides internal users at Continuum, I don't know of any other general communities.

**Comment by msarahan** [comment]
Please confirm that with @ilanschnell or write a script to check.  I'm less concerned about user communities, but we absolutely must not give enterprise customers any cause for concern.  User communities can chime in here if they have issues.

**Comment by kalefranz** [comment]
@msarahan Did a search of all our anaconda recipes, and confirmed myself with Ilan.  The *only* use is `mpi4py/pre-link.bat`.  @ilanschnell is changing that to a post-link script for the next time it's built.

**Comment by sbourdeauducq** [comment]
Can you please explain why this warning shows up when installing this package:
https://github.com/m-labs/conda-recipes/blob/master/conda/outputcheck/meta.yaml
We're not using any pre-link script here...

**Comment by nehaljwani** [comment]
@sbourdeauducq If you are referring to the following...
```bash
Installing local::outputcheck-0.4.2-py_7 requires a minimum conda version of 4.3.
```
...then you are probably using conda 4.2.x and you are receiving that message because of the following [line](https://github.com/m-labs/conda-recipes/blob/master/conda/outputcheck/meta.yaml#L11) in the recipe (in the build section):
```
  noarch: python
```

**Comment by sbourdeauducq** [comment]
OK. Eventually I figured that out; but it is quite obscure and maybe the error message could provide a hint about those old-style noarch packages.

**Pull request merged**